quinn-proto/src/connection/mod.rs

```diff
@@ -3022,7 +3022,7 @@ impl Connection {
         &mut self,
         now: Instant,
         space_id: SpaceId,
-        buf: &mut Vec<u8>,
+        buf: &mut (impl BufMut + BufLen),
```

Comment:
If there were already a single externally defined trait that could be fully sufficient for replacing `&mut Vec<u8>`/`&mut TransmitBuf` with `&mut (impl ThatTrait)`, I would enjoy doing so. But in this case where 1. we are taking on the complexity of defining an additional `BufLen` trait and its impl blocks 2. we are using compound `&mut (impl X + Y)` blocks to utilize it and 3. these methods are nevertheless only called with a one implementation, I'm don't really think it's worth it compared to just passing `&mut Vec<u8>` or `&mut TransmitBuf`.